src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@ public interface MachineService {
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<? extends Machine> getAll(Class<? extends T> type);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to store the result before returning it. This can make debugging easier and the code more readable.